thirdparty/build-thirdparty.sh

```diff
@@ -50,8 +50,264 @@ if [ ! -f ${TP_DIR}/vars.sh ]; then
 fi
 . ${TP_DIR}/vars.sh
 
+# Check args
+usage() {
+    echo "
+Usage: $0 [options...] [packages...]
+
+  Description:
+    Build thirdparty dependencies for StarRocks. If no packages are specified,
+    all packages will be built in the default order.
+
+  Optional options:
+    -j<num>                Build with <num> parallel jobs (can also use -j <num>)
+    --clean                Clean extracted source before building
+    --continue <package>   Continue building from specified package
+    -h, --help             Show this help message
+
+  Examples:
+    # Build all packages with default parallelism
+    $0
+
+    # Build all packages with 8 parallel jobs
+    $0 -j8
+
+    # Clean and rebuild everything with 16 parallel jobs
+    $0 --clean -j16
+
+    # Continue building from rocksdb (useful after build failure)
+    $0 --continue rocksdb
+
+    # Build only specific packages (in dependency order)
+    # Note: packages are built in the same order as full build to respect dependencies
+    $0 openssl curl protobuf
+
+    # Clean and build only specific packages
+    $0 --clean boost thrift
+  "
+    exit 1
+}
+
+get_all_package_sources() {
+    local sources=(
+        "$LIBEVENT_SOURCE"
```

Comment:
these long list can be easily out of sync with the actual package list, better to use `$TP_ARCHIVES` to create the list.